ches/index.ts

```diff
@@ -1,10 +1,113 @@
-import dotenv from 'dotenv';
-dotenv.config();
-import ChesService from './chesService';
-
-export default new ChesService({
-  tokenUrl: process.env.CHES_TOKEN_URL || 'default',
-  clientId: process.env.CHES_CLIENT_ID || 'default',
-  clientSecret: process.env.CHES_CLIENT_SECRET || 'default',
-  apiUrl: process.env.CHES_API_URL || 'default',
-});
+import compact from 'lodash.compact';
+import uniq from 'lodash.uniq';
+import castArray from 'lodash.castarray';
+
+type NullOrString = string | null | undefined;
+type EmailAddress = string | undefined;
+
+interface Email {
+  bodyType?: 'html' | 'text';
+  from?: string;
+  subject: string;
+  body: string;
+  to: EmailAddress[];
+  bcc?: EmailAddress[];
+  cc?: EmailAddress[];
+  delayTS?: number;
+  encoding?: 'base64' | 'binary' | 'hex' | 'utf-8';
+  priority?: 'normal' | 'low' | 'high';
+  tag?: string;
+}
+
+interface TokenData {
+  tokenUrl: string;
+  clientId: string;
+  clientSecret: string;
+}
+
+const safeEmails = (emails: Array<NullOrString>): string[] => uniq(compact(castArray(emails)));
+
+const getToken = async ({ tokenUrl, clientId, clientSecret }: TokenData): Promise<string | null> => {
+  try {
+    const response = await fetch(tokenUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: new URLSearchParams({
+        grant_type: 'client_credentials',
+        client_id: clientId,
+        client_secret: clientSecret,
+      }),
+    });
+
+    if (!response.ok) {
+      console.error('Error retrieving token:', response.statusText);
+      return null;
+    }
+
+    const data = (await response.json()) as { access_token: string };
+    return data.access_token;
+  } catch (error) {
+    console.error('Exception retrieving token:', error);
+    return null;
+  }
+};
+
+const sendEmail = async (email: Email): Promise<void> => {
```

Comment:
I kind of prefer the `object destruction way of retrieving each property`, but could be a personal choice.
https://github.com/bcgov/platform-services-registry/pull/1306/files#diff-176385a08bc07b6d37488033a729d5ddd96392de93a65659234ef593221c0e10L68